### DISCUSSION TITLE
Resolving ambiguities between object-fit and object-position

### DISCUSSION THREAD

**Discussion by AndreasFaust** [tailwindlabs/tailwindcss#14540] [Help] [answered]
I am trying to combine object-fit and -position with CSS-variables:
```html
  <div 
    class="object-[position:--position] object-[string:--fit]"
    style="--fit: cover; --position: center;"
  />
```
I’m trying to distinguish them by adding CSS-data-types. 
Unfortunately it does not work. Is this a bug or am I doing something wrong?

**Comment by wongjn** [comment, marked as answer]
[Arbitrary value syntax](https://tailwindcss.com/docs/adding-custom-styles#using-arbitrary-values) is only available for ["dynamic-value" utilities](https://tailwindcss.com/docs/plugins#dynamic-utilities). The `object-fit`-related class names [are](https://github.com/tailwindlabs/tailwindcss/blob/ed3c5356b77b6c428e9c0ea5bc32719dcdf94128/src/corePlugins.js#L2023-L2029) [static utilities](https://tailwindcss.com/docs/plugins#static-utilities).

For your use case, you might as well fully use inline `style` like:

```html
<div style="object-fit: cover; object-position: center">
```

Otherwise, if there is a need for inheritance of the CSS variables, you could do:

```html
<div style="object-fit: var(--fit); object-position: var(--position); --fit: cover; --position: center;">
```
Or
```html
<div
  class="[object-fit:var(--fit)] [object-position:var(--position)]"
  style="--fit: cover; --position: center;"
>
```

**Comment by AndreasFaust** [comment]
Thank you for the quick answer, @wongjn !
I oversimplyfied my example. The reason why I am using CSS-variables is to address breakpoints:
```html
<div
  class="sm:[object-fit:var(--fit-mobile)] lg:[object-fit:var(--fit-desktop)] sm:[object-position:var(--position-mobile)] lg:[object-position:var(--position-desktop)]"
  style="--fit-mobile: cover; --fit-desktop: contain; --position-mobile: center; --position-desktop: top;"
>
```
Writing the whole property as arbitrary value works perfectly. Thank you again!
